tools/mtmd/clip.cpp

```diff
@@ -847,13 +849,19 @@ struct clip_graph {
             int n_embd = clip_n_mmproj_embd(ctx);
             const int d_head = 128;
             int n_head = n_embd/d_head;
+            // Use actual config value if available, otherwise fall back to hardcoded values
             int num_query = 96;
-            if (ctx->model.hparams.minicpmv_version == 2) {
-                num_query = 96;
-            } else if (ctx->model.hparams.minicpmv_version == 3) {
-                num_query = 64;
-            } else if (ctx->model.hparams.minicpmv_version == 4) {
-                num_query = 64;
+            if (ctx->model.hparams.minicpmv_query_num > 0) {
+                num_query = ctx->model.hparams.minicpmv_query_num;
+            } else {
+                // Fallback to hardcoded values for legacy models
+                if (ctx->model.hparams.minicpmv_version == 2) {
+                    num_query = 96;
+                } else if (ctx->model.hparams.minicpmv_version == 3) {
+                    num_query = 64;
+                } else if (ctx->model.hparams.minicpmv_version == 4) {
+                    num_query = 64;
+                }
```

Comment:
move this into `clip_model_loader` and populate `ctx->model.hparams.minicpmv_query_num` when loading the model